support/cas-server-support-x509-core/src/main/java/org/apereo/cas/adaptors/x509/authentication/principal/X509SubjectAlternativeNameUPNPrincipalResolver.java

```diff
@@ -141,7 +163,26 @@ protected String resolvePrincipalInternal(final X509Certificate certificate) {
             }
         } catch (final CertificateParsingException e) {
             LOGGER.error("Error is encountered while trying to retrieve subject alternative names collection from certificate", e);
-            return null;
+            return getFallbackPrincipal(certificate);
+        }
+
+        return getFallbackPrincipal(certificate);
+    }
+
+    private String getFallbackPrincipal(final X509Certificate certificate) {
```

Comment:
While this is OK, you can see that you are sort of cascading inward where the if statements are taking the shape of a diamond :) An easier way to short-circuit this which argubly might be easier on the eye and follow-up reads would be:

```java
if (StringUtils.isEmpty(fallbackPrincipalAttributeName)) {
  return whatever;
}


val attributes = extractPersonAttributes(certificate);
val attribute = attributes.get(fallbackPrincipalAttributeName);
if (attribute == null) {
   return whatever;
}
...
```

Avoid the diamond.